### PR TITLE
Change to LGPL v3 license

### LINKED ISSUE #1763: License change to GMP-compatible license
It would be nice to be able to copy GMP- and MPFR-code for use in FLINT. For example, it would be advantageous to use GMP's `mpn_addmul_1` assembly code to generate `mpn_mulhigh_basecase`, used for fast floating point arithmetic and integer division.

### PR DESCRIPTION
**Pull request by albinahlback**
Resolves #1763

### CODE REVIEW

File: doc/source/introduction.rst [hunk 1]
```diff
@@ -66,6 +66,9 @@ files.
 License
 -----------------------------------------------------------------------------
 
-FLINT is distributed under the LGPL License, version 2.1+. There is a copy
-of the license included in the repository and distribution tarballs.
+FLINT is distributed under the GNU Lesser General Public License (LGPL)
+version 3. There is a copy of the license included in the repository and
+distribution tarballs.
 
+Note, however, that before FLINT version 3.1, FLINT was distributed under LGPL
```
Comment (fredrik-johansson) [inline doc/source/introduction.rst:73]: LGPL2 was used starting from 2.6. Before that, FLINT was GPL2.

File: doc/source/introduction.rst [hunk 2]
```diff
@@ -66,6 +66,9 @@ files.
 License
 -----------------------------------------------------------------------------
 
-FLINT is distributed under the LGPL License, version 2.1+. There is a copy
-of the license included in the repository and distribution tarballs.
+FLINT is distributed under the GNU Lesser General Public License (LGPL)
+version 3. There is a copy of the license included in the repository and
```
Comment (fredrik-johansson) [inline doc/source/introduction.rst:70]: version 3 or later

File: README.md [hunk 3]
```diff
@@ -61,6 +60,5 @@ The current maintainers are:
 
 ## License
 
-FLINT is distributed under LGPL 2.1 (GNU Lesser General Public License).
-See the LICENSE file.
-
+FLINT is distributed under LGPL 3 (GNU Lesser General Public License). See the
```
Comment (fredrik-johansson) [inline README.md:63]: version 3 or later

**Comment by albinahlback** [comment]
I think this is ready to go.

**Comment by fredrik-johansson** [comment]
Yep, LGTM.

**Comment by albinahlback** [comment]
Just fixed licenses in the new files from #1766.

**Comment by fredrik-johansson** [comment]
Thanks!

**Pull request merged**